Source/Plugins/Core/com.equella.core/js/tsrc/search/components/SearchResultAttachmentsList.tsx

```diff
@@ -0,0 +1,342 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  Accordion,
+  AccordionDetails,
+  AccordionSummary,
+  Badge,
+  Grid,
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemSecondaryAction,
+  ListItemText,
+  Theme,
+  Typography,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import Tooltip from "@material-ui/core/Tooltip";
+import AttachFile from "@material-ui/icons/AttachFile";
+import DragIndicatorIcon from "@material-ui/icons/DragIndicator";
+import ExpandMore from "@material-ui/icons/ExpandMore";
+import InsertDriveFile from "@material-ui/icons/InsertDriveFile";
+import Search from "@material-ui/icons/Search";
+import { SyntheticEvent, useEffect, useState } from "react";
+import * as React from "react";
+import ItemAttachmentLink from "../../components/ItemAttachmentLink";
+import { LightboxConfig } from "../../components/Lightbox";
+import {
+  getSearchPageAttachmentClass,
+  isSelectionSessionInSkinny,
+  isSelectionSessionInStructured,
+  isSelectionSessionOpen,
+  selectResource,
+} from "../../modules/LegacySelectionSessionModule";
+import {
+  determineAttachmentViewUrl,
+  determineViewer,
+  ViewerDefinition,
+} from "../../modules/ViewerModule";
+import { languageStrings } from "../../util/langstrings";
+import { ResourceSelector } from "./ResourceSelector";
+import * as OEQ from "@openequella/rest-api-client";
+
+const {
+  searchResult: searchResultStrings,
+  selectResource: selectResourceStrings,
+} = languageStrings.searchpage;
+
+const useStyles = makeStyles((theme: Theme) => ({
+  nested: {
+    paddingLeft: theme.spacing(4),
+  },
+  attachmentExpander: {
+    marginTop: theme.spacing(2),
+    marginBottom: theme.spacing(2),
+  },
+  attachmentBadge: {
+    backgroundColor: theme.palette.background.paper,
+    color: theme.palette.secondary.main,
+    borderRadius: "50%",
+  },
+  attachmentListItem: {
+    width: "100%",
+  },
+}));
+
+export interface AttachmentAndViewer {
+  /**
+   * The details of an attachment.
+   */
+  attachment: OEQ.Search.Attachment;
+  /**
+   * Viewer information for the attachment, including which viewer to be used and the view URL.
+   */
+  viewer: ViewerDefinition;
+}
+
+export interface SearchResultAttachmentsListProps {
+  item: OEQ.Search.SearchResultItem;
+  getViewerDetails: (
+    mimeType: string
+  ) => Promise<OEQ.MimeType.MimeTypeViewerDetail>;
+  /**
+   * Error handler for standard management of errors during processing - especially comms errors.
+   */
+  handleError: (error: Error) => void;
+}
+export const SearchResultAttachmentsList = ({
```

Comment:
Please have a blank line around definitions of functions.

```suggestion

export const SearchResultAttachmentsList = ({
```